src/Features/CSharp/Portable/InlineParameterNameHints/InlineParamNameHintsService.cs

```diff
@@ -0,0 +1,109 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Composition;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CSharp.Extensions;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Host.Mef;
+using Microsoft.CodeAnalysis.InlineParameterNameHints;
+using Microsoft.CodeAnalysis.Text;
+#nullable enable
+
+namespace Microsoft.CodeAnalysis.CSharp.InlineParameterNameHints
+{
+    /// <summary>
+    /// The service to locate the positions in which the adornments should appear
+    /// as well as associate the adornments back to the parameter name
+    /// </summary>
+    [ExportLanguageService(typeof(IInlineParamNameHintsService), LanguageNames.CSharp), Shared]
+    internal class InlineParamNameHintsService : IInlineParamNameHintsService
```

Comment:
somewhat surprised this doesn't derive from an abstract base where C#/VB shared logic woudl live.   will see if that matters as review continues.

Reply:
As it stands this is all doing syntactic testing, my gut is at this time we won't really have much to share...